RELEASE_NOTES.md

```diff
@@ -8,6 +8,10 @@ This document contains release notes for the current major release and all patch
 For prior releases,
 see [PRIOR\_RELEASE\_NOTES.md](PRIOR_RELEASE_NOTES.md).
 
+## 12.17.0 (2017-10-10)
```

Comment:
Minor note: This will actually go out as 12.17.1 or greater because of expeditor.  We can potentially avoid the version bump with the "Version: Skip Bump" tag, but I'm unsure if that will end up skipping any other steps.  Might be worth trying just to find out.

I think in the long run, we might want to reconsider how we manage this file.